be/src/olap/comparison_predicate.cpp

```diff
@@ -145,28 +146,68 @@ COMPARISON_PRED_COLUMN_BLOCK_EVALUATE(LessEqualPredicate, <=)
+                    for (uint16_t i = 0; i < *size; i++) {                                         \
+                        uint16_t idx = sel[i];                                                     \
+                        sel[new_size] = idx;                                                       \
+                        const auto& cell_value =                                                   \
+                                reinterpret_cast<const vectorized::Int32&>(data_array[idx]);       \
+                        bool ret = !null_bitmap[idx] && (cell_value OP code);                      \
+                        new_size += _opposite ? !ret : ret;                                        \
+                    }                                                                              \
+                }                                                                                  \
+            } else {                                                                               \
+                auto* nested_col_ptr =                                                             \
+                        vectorized::check_and_get_column<vectorized::PredicateColumnType<type>>(   \
+                                nested_col);                                                       \
+                auto& data_array = nested_col_ptr->get_data();                                     \
+                for (uint16_t i = 0; i < *size; i++) {                                             \
+                    uint16_t idx = sel[i];                                                         \
+                    sel[new_size] = idx;                                                           \
+                    const type& cell_value = reinterpret_cast<const type&>(data_array[idx]);       \
+                    bool ret = !null_bitmap[idx] && (cell_value OP _value);                        \
+                    new_size += _opposite ? !ret : ret;                                            \
+                }                                                                                  \
             }                                                                                      \
             *size = new_size;                                                                      \
+        } else if (column.is_column_dictionary()) {                                                \
+            if constexpr (std::is_same_v<type, StringValue>) {                                     \
+                auto& dict_col =                                                                   \
+                        reinterpret_cast<vectorized::ColumnDictionary<vectorized::Int32>&>(column);\
+                auto& data_array = dict_col.get_data();                                            \
+                auto code = dict_col.find_code(_value);                                            \
```

Comment:
we better use `std::move`,  because dict_col.find_code return a large local var.
`auto code = std::move(dict_col.find_code(_value));  `

Reply:
For function return value initialized variables,  can rely on the compiler's RVO optimization. 
Even create an int value doesn't cost much. Because it is outside the loop, temporary objects are not created frequently.
just my opinion.